### PR TITLE
write to the stub outside the modifying function

### PR DESCRIPTION
**Pull request by relud**
r? @oremj

### CODE REVIEW

**Comment by relud** [comment]
tested against fake stub installer to behave the same as original code

File: modify.py [hunk 1]
```diff
@@ -47,23 +48,30 @@ def write_attribution_data(filepath, data):
...
+        return False
 
-if __name__ == "__main__":
-    write_attribution_data(sys.argv[1], sys.argv[2])
 
+if __name__ == "__main__":
+    # read mapped installer as bytearray so it can be modified
+    with open(sys.argv[1], 'r+b') as f:
+        mapped = bytearray(f.read())
+    write_attribution_data(mapped, sys.argv[2])
```
Comment (oremj) [inline modify.py]: Might be better to wrap the `write_attribution_data` call in a try block instead of the code within the function.

Comment (relud) [inline modify.py]: sure, but that would have made the diff harder to read. i actually want to remove the try block entirely and just let it throw the exception.

**Comment by relud** [comment]
try block removed

**Comment by oremj** [comment]
lgtm

**Pull request merged**